bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/DigitalSTROMHandlerFactory.java

```diff
@@ -256,8 +256,9 @@ private boolean checkUserPassword(Configuration configuration) {
     protected synchronized void removeHandler(ThingHandler thingHandler) {
         if (thingHandler instanceof BridgeHandler) {
             String uid = thingHandler.getThing().getUID().getAsString();
-            if (discoveryServiceManagers.get(uid) != null) {
-                discoveryServiceManagers.get(uid).unregisterDiscoveryServices(bundleContext);
+            DiscoveryServiceManager discoveryServiceManager = discoveryServiceManagers.get(uid);
+            if (discoveryServiceManager != null) {
+                discoveryServiceManager.unregisterDiscoveryServices(bundleContext);
                 discoveryServiceManagers.remove(uid);
             }
```

Comment:
```suggestion
            DiscoveryServiceManager discoveryServiceManager = discoveryServiceManagers.remove(uid);
            if (discoveryServiceManager != null) {
                discoveryServiceManager.unregisterDiscoveryServices(bundleContext);
            }
```

Also you should change `discoveryServiceManagers` to a `ConcurrentHashMap`